config/config.go

```diff
@@ -539,7 +544,15 @@ func (c ControllerSettings) MaxControllerCount() int {
 
 func (c ControllerSettings) ControllerRollingUpdateMinInstancesInService() int {
 	if c.AutoScalingGroup.RollingUpdateMinInstancesInService == 0 {
-		return c.MaxControllerCount() - 1
+
+		maxControllerCount := c.MaxControllerCount()
+
+		//never accept controller size = 0
+		if maxControllerCount > 0 {
+			return maxControllerCount - 1
+		}
+
+		return 1
```

Comment:
Just curious but would you also like to make the minimum number of controller nodes returned by `MinControllerCount()` to 1?
If you agree, I'll make it so.